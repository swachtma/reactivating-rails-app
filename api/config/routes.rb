Rails.application.routes.draw do
  scope :api do
    get 'github/', to: "authentication#github", format: false
    get 'hydrate_user/', to: "authentication#show"
    get 'nodes/', to: "nodes#index"
    get 'chapters/', to: "chapters#index"
    resources :bookmarks, only: [:index, :create]
  end
end