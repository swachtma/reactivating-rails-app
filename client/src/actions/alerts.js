import * as constants from "../constants/alerts";

export const addAlert = (message, level = constants.lvls[0], scope = constants.scopes[0]) => ({
  type: constants.ADD_ALERT, payload: {message, level, scope, timestamp: Date.now()}
});

export const clearAlerts = () => ({ type: constants.CLEAR_ALERTS }) ;