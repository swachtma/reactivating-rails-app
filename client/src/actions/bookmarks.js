import * as constants from "../constants/bookmarks";

export const setBookmarkLocations = (chapter_id) => ({
  type: constants.SET_BOOKMARK_LOCATIONS, payload: {chapter: chapter_id}
});

export const setBookmarkOffered = (bool) => ({
  type: constants.SET_BOOKMARK_OFFERED, payload: {prompt_shown: bool}
});