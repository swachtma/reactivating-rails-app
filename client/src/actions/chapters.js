import * as constants from '../constants/chapters'

export const loadChapters = (payload) => ({ type: constants.LOAD_CHAPTERS, payload })
