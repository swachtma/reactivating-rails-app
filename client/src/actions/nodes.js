import * as constants from '../constants/nodes'

export const loadNodes = (payload) => ({ type: constants.LOAD_NODES, payload })
