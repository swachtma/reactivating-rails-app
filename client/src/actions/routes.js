import * as routes from '../constants/settings';

export const routeHome = () => ({type: routes.HOME_ROUTE});

export const routeChapter = (chapter_id) => (
  {type: routes.CHAPTER_ROUTE, payload: { chapter_id }}
);