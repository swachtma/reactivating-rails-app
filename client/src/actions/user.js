import * as constants from '../constants/user'

export const setUser = (payload) => ({
  type: constants.SET_USER, payload
})

export const signalSignout = () => ({ type: constants.SIGNAL_SIGNOUT })
export const clearUser = () => ({ type: constants.CLEAR_USER })
