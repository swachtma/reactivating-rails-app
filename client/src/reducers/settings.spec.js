/* global expect */
import  settingsReducer from './settings';
import * as actions from '../actions/routes';

let defaultState = {};

describe("Chapter & Home Routes", ()=>{
  test("Assign default chapter of one on HOME_ROUTE", ()=>{
    expect(
      settingsReducer(defaultState,actions.routeHome())
    ).toMatchSnapshot();
  });
  
  test("assign payload chapter on CHAPTER_ROUTE",()=>{
    expect(
      settingsReducer(defaultState,actions.routeChapter(5))
    ).toMatchSnapshot();
  });
});