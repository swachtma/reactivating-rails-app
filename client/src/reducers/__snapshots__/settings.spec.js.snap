// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chapter & Home Routes Assign default chapter of one on HOME_ROUTE 1`] = `
Object {
  "active_chapter_id": 1,
}
`;

exports[`Chapter & Home Routes assign payload chapter on CHAPTER_ROUTE 1`] = `
Object {
  "active_chapter_id": 5,
}
`;
